server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/Main.scala

```diff
@@ -21,58 +16,38 @@ import scala.concurrent.ExecutionContext
 
 object Main extends IOApp {
   override def run(args: List[String]): IO[ExitCode] = {
-    implicit val unsafeLogger = Slf4jLogger.getLogger[IO]
+    implicit val logger = Slf4jLogger.getLogger[IO]
 
     val app: Stream[IO, Unit] = for {
       blockingEc <- Stream.resource[IO, ExecutionContext](ExecutionContexts.cachedThreadPool)
       implicit0(it: Linebacker[IO]) <- Stream.eval(Linebacker.bounded(Linebacker.fromExecutionContext[IO](blockingEc), 255))
       appConfig <- Stream.fromEither[IO](Config.appConfig)
-      storageLinksCache <- cachedResource[Path, StorageLink](
+      storageLinksCache <- cachedResource[RelativePath, StorageLink](
         appConfig.pathToStorageLinksJson,
         blockingEc,
         storageLink => List(storageLink.localBaseDirectory.path -> storageLink, storageLink.localSafeModeBaseDirectory.path -> storageLink)
       )
-      metadataCache <- cachedResource[Path, AdaptedGcsMetadataCache](
+      metadataCache <- cachedResource[RelativePath, AdaptedGcsMetadataCache](
         appConfig.pathToGcsMetadataJson,
         blockingEc,
-        metadata => List(metadata.localPath.asPath -> metadata)
+        metadata => List(metadata.localPath -> metadata)
       )
       googleStorageService <- Stream.resource(GoogleStorageService.fromApplicationDefault())
       storageLinksService = StorageLinksService(storageLinksCache)
       googleStorageAlg = GoogleStorageAlg.fromGoogle(GoogleStorageAlgConfig(appConfig.objectService.workingDirectory), googleStorageService)
       storageLinkAlg = StorageLinksAlg.fromCache(storageLinksCache)
       objectService = ObjectService(appConfig.objectService, googleStorageAlg, blockingEc, storageLinkAlg, metadataCache)
-      server <- BlazeServerBuilder[IO].bindHttp(appConfig.serverPort, "0.0.0.0").withHttpApp(WelderApp(objectService, storageLinksService).service).serve
+      serverStream = BlazeServerBuilder[IO].bindHttp(appConfig.serverPort, "0.0.0.0").withHttpApp(WelderApp(objectService, storageLinksService).service).serve
+      cleanUpCache = BackgroundTask.cleanUpLock(metadataCache, appConfig.cleanUpLockFrequency)
+      _ <- Stream(cleanUpCache, serverStream.drain).covary[IO].parJoin(2)
     } yield ()
 
     app
       .handleErrorWith { error =>
         Stream.eval(Logger[IO].error(error)("Failed to start server")) >> Stream.raiseError[IO](error)
       }
-      .evalMap(_ => IO.never)
```

Comment:
Why don't we need this anymore? Is it because we now have the infinite `BackgroundTask` stream?

Reply:
yep...the background task will prevent the app from exiting, so we don't really need this...The background task won't actually work with this because background process is an infinite Stream, and this will prevent the first emitted IO to not return.